#! /bin/sh -e
## 01_arch_fallback.dpatch by Eduard Bloch <blade@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Using dpkg-architecture output to declare names of unsupported arches

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@
Index: assembly.c
===================================================================
RCS file: /cvs/public/mono/mono/metadata/assembly.c,v
retrieving revision 1.104
diff -u -r1.104 assembly.c
--- a/mono/metadata/assembly.c	16 Jun 2004 11:48:32 -0000	1.104
+++ b/mono/metadata/assembly.c	16 Jun 2004 18:59:01 -0000
@@ -655,7 +655,8 @@
 
 	ass = mono_assembly_load_from (image, fname, status);
 
-	mono_config_for_assembly (ass->image);
+	if (ass)
+		mono_config_for_assembly (ass->image);
 
 	g_free (fname);
 
